src/common/options.cc

```diff
@@ -2567,6 +2567,11 @@ std::vector<Option> get_global_options() {
     .set_default(100)
     .set_description(""),
 
+    Option("osd_pg_max_per_osd", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    .set_default(300)
```

Comment:
This seems like a rather low limit for a hard restriction that will stop pools functioning.

If it's meant to be a sanity thing to prevent completely overwhelming an OSD, then maybe something in the thousands is more appropriate?

Reply:
Also, maybe this should be a per-TB setting?  In a heterogeneous cluster it may not make sense to apply the same PG limit to a 10TB drive as a 1TB drive.

Reply:
make sense to me! but per http://docs.ceph.com/docs/master/rados/operations/placement-groups/, 300 is already a tolerant setting. how about 600?

Reply:
will update `doc/rados/configuration/pool-pg-config-ref.rst` once we reach an agreement on this.

Reply:
I'd be okay with 600 or 1000. 

Note that we also have https://github.com/ceph/ceph/pull/17427 in flight.  That will prevent pool creation once you hit a limit and trigger a warning if it is exceeded.

Maybe this hard limit should be called osd_pg_hard_max_per_osd?  Or maybe it should be a factor of the warning/"can't create a pool" limit (which is softer)?  (Like, 2x that and the osds will refuse to instantiate osds?)